src/query.ts

```diff
@@ -0,0 +1,526 @@
+// @deno-types="../build/sqlite.d.ts"
+import { StatementPtr, Wasm } from "../build/sqlite.js";
+import { getStr, setArr, setStr } from "./wasm.ts";
+import { Status, Types, Values } from "./constants.ts";
+import { SqliteError } from "./error.ts";
+
+/**
+ * The default type for returned rows.
+ */
+// deno-lint-ignore no-explicit-any
+export type Row = Array<any>;
```

Comment:
Should this be `type Row = Array<unknown>` instead?

Reply:
(The benefit being that it encourages people to type-annotate their queries)

Reply:
I think its best to stick with `any`. In most cases, we _know_ what the type is, we just have to cast it. Unknown is typically when you _dont_ know the type and need to parse it. 
```ts
const row: PhoneNumber = stmt.queryOne() as any
```

This feels very similar to the discussion that the official typescript repo had regarding `JSON.parse`. Ultimately they chose to keep it returning `any`, and let developers wrap it in another function if they wished to use `unknown`

Reply:
Hm the thing with the new api is that you can easily annotate the row the type, which is preferable especially for prepared queries since it’s that means you can annotate the type next to the query and then have it checked everywhere the query is used:

```typescript
const query = dB.prepareQuery<[number, string]>("SELECT id, name FROM users");
// …
for (const [id, name] of query.iter()) {
// …
}
```

My hope was that using unknown would encourage annotating the type, which would then mean the type annotation is always close to where the query SQL is defined as well 🤔

Reply:
(or type annotation for the top level query method without a prepare)
```typescript
const users = db.query<[number, string]>("SELECT id, name FROM users");
```

Reply:
@andykais I was trying to find the TypeScript discussion you mentioned, but instead I stumbled across [this thread](https://github.com/microsoft/TypeScript/issues/10715) around `unknown`, which cites `JSON.parser` as one of the motivations for the `unknown` type.

Especially since the API we provide has an explicit type parameter for annotating `query` and `preparedQuery`, I think using `unknown` is a win overall, since it's the safer default and encourages co-locating type definitions with the query they depend on (vs having `any` put the burden on annotating/ checking the type everywhere the prepared query is used).